### PR TITLE
Transform Dynamodb response for easy consumption

### LINKED ISSUE #3010: [Feature] Option to auto-transform DynamoDB responses for easy consumption
## Summary

Currently, DynamoDB sends raw response from AWS to the client. But this format is not very convenient to use within Appsmith and requires a tedious transformation to be applied. If such a routine transformation can be done by Appsmith the data would be readily consumable by the client widgets.

Example raw response data (of `Scan` operation):

```
{
    "Items":[
        0:{
            "createdAt":{
                "SS":[
                ]
                "BS":[
                ]
                "S":"1553254864424"
                "NS":[
                ]
                "L":[
                ]
                "M":{
                }
            }
            "name":{
                "SS":[
                ]
                "BS":[
                ]
                "S":"internet"
                "NS":[
                ]
                "L":[
                ]
                "M":{
                }
            }
        }
    ]
    "Count":4
    "ScannedCount":4
    "LastEvaluatedKey":{
    }
}
```

This can be simplified to:

```
{
    "Items":[
        0:{
            "createdAt":"1553254864424"
            "name":"internet"
        }
    ]
    "Count":4
    "ScannedCount":4
    "LastEvaluatedKey":{
    }
}
```

### PR DESCRIPTION
**Pull request by sumitsum**
## Description
- add support to handle dnyamoDb Binary type
- transform raw response to show extracted response and raw response for:
GetItem
BatchGetItem
DeleteItem
PutItem
Scan
TransactGetItems
UpdateItem
-  modify TC. 

Fixes #3010 

## Type of change
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

## How Has This Been Tested?
Updated View:
Screenshot 2021-02-18 at 9 01 54 PM

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-plugins/dynamoPlugin/src/main/java/com/external/plugins/DynamoPlugin.java [hunk 1]
```diff
@@ -50,6 +50,12 @@
 
 public class DynamoPlugin extends BasePlugin {
 
+    private static final String SCAN_ACTION_VALUE = "Scan";
+    private static final String ITEMS_KEY = "Items";
+    private static final String TRANSFORMED_DATA_LABEL = "value";
+    private static final String RAW_DATA_LABEL = "raw";
+    private static final String S_FIELD_LABEL = "S";
```
Comment (mohanarpit) [inline app/server/appsmith-plugins/dynamoPlugin/src/main/java/com/external/plugins/DynamoPlugin.java:57]: Where are we handling the other data types like number, boolean, lists, etc.? In the DynamoDB documentation, I can see that there are multiple data types that can be returned.

**Comment by mohanarpit** [comment]
Also, I noticed that the response structure is not how it's defined in the issue. Can you please conform to that structure? We should put the raw response in a different object. Not with each key. Also, we should remove the "0", "1" etc in the array. That's not useful as long as we maintain the order as given in the DynamoDB response.

For example:

```
{
   "raw": <The raw response from DynamoDB here>,
    "Items":[
        {
            "createdAt":"1553254864424"
            "name":"internet"
        },
        {
            "createdAt":"1553254864426"
            "name":"newInternet"
        }
    ]
    "Count":4
    "ScannedCount":4
    "LastEvaluatedKey":{
    }
}
```

File: app/server/appsmith-plugins/dynamoPlugin/src/main/java/com/external/plugins/DynamoPlugin.java (hunk 1 above)
Comment (sumitsum) [reply app/server/appsmith-plugins/dynamoPlugin/src/main/java/com/external/plugins/DynamoPlugin.java]: Done.

**Comment by sumitsum** [comment]
> Also, I noticed that the response structure is not how it's defined in the issue. Can you please conform to that structure?
> 
> We should put the raw response in a different object. Not with each key. Also, we should remove the "0", "1" etc in the array. That's not useful as long as we maintain the order as given in the DynamoDB response.
> 
> For example:
> 
> ```
> {
>    "raw": <The raw Items response object from DynamoDB here>,
>     "Items":[
>         {
>             "createdAt":"1553254864424"
>             "name":"internet"
>         },
>         {
>             "createdAt":"1553254864426"
>             "name":"newInternet"
>         }
>     ],
>     "Count":4
>     "ScannedCount":4
>     "LastEvaluatedKey":{
>     }
> }
> ```

Done.

File: app/server/appsmith-plugins/dynamoPlugin/src/main/java/com/external/plugins/DynamoPlugin.java [hunk 2]
```diff
@@ -71,6 +91,148 @@ public DynamoPlugin(PluginWrapper wrapper) {
...
+                                        ". Please reach out to Appsmith customer support to resolve this."
+                        );
+                }
+
+                for (Map.Entry<String, Object> responseEntry: rawResponse.entrySet()) {
+                    if (!responseEntry.getKey().equals(topLevelKey)) {
+                        transformedResponse.put(responseEntry.getKey(), responseEntry.getValue());
+                    }
+                    else {
+                        if (action.equals(SCAN_ACTION_VALUE)) {
```
Comment (mohanarpit) [inline app/server/appsmith-plugins/dynamoPlugin/src/main/java/com/external/plugins/DynamoPlugin.java:200]: Minor: As a best practice, always have the constant variable first. This ensures that we don't run the risk of NPEs across the code base.
```suggestion
                        if (SCAN_ACTION_VALUE.equals(action)) {
```

Comment (sumitsum) [reply app/server/appsmith-plugins/dynamoPlugin/src/main/java/com/external/plugins/DynamoPlugin.java]: Done.

**Review by mohanarpit — APPROVED** [review]

**Pull request merged**